editor/src/components/inspector/sections/component-section/component-section.tsx

```diff
@@ -496,6 +497,81 @@ const RowForUnionControl = betterReactMemo(
   },
 )
 
+interface RowForVectorControlProps extends AbstractRowForControlProps {
+  controlDescription: ObjectControlDescription
+}
+
+const RowForVectorControl = betterReactMemo(
+  'RowForVectorControl',
+  (props: RowForVectorControlProps) => {
+    const { propPath, controlDescription, isScene } = props
+    const title = titleForControl(propPath, controlDescription)
+    const propMetadata = useComponentPropsInspectorInfo(propPath, isScene, controlDescription)
+    const contextMenuItems = Utils.stripNulls([
+      addOnUnsetValues([PP.toString(propPath)], propMetadata.onUnsetValues),
+    ])
+    return (
+      <>
+        <InspectorContextMenuWrapper
+          id={`context-menu-for-${PP.toString(propPath)}`}
+          items={contextMenuItems}
+          data={null}
+        >
+          <UIGridRow padded={true} variant='<---1fr--->|------172px-------|'>
+            {title}
+            <UIGridRow padded={false} variant='<--1fr--><--1fr--><--1fr-->'>
+              {mapToArray((innerControl: ControlDescription, prop: string) => {
+                const innerPropPath = PP.appendPropertyPathElems(propPath, [prop])
+                const propName = `${PP.lastPart(propPath)}`
+                const innerMetadata = useComponentPropsInspectorInfo(
+                  innerPropPath,
+                  isScene,
+                  innerControl,
+                )
+                return (
+                  <ControlForNumberProp
+                    key={`object-control-row-${PP.toString(innerPropPath)}`}
+                    controlDescription={innerControl as NumberControlDescription}
+                    propName={propName}
+                    propMetadata={innerMetadata}
+                  />
+                )
+              }, controlDescription.object)}
+            </UIGridRow>
+          </UIGridRow>
+        </InspectorContextMenuWrapper>
+      </>
+    )
+  },
+)
+
+function isVectorControlDescription(controlDescription: ObjectControlDescription): boolean {
```

Comment:
have you considered adding dedicated Vector3 (and vector2) control types? (adding it here https://github.com/concrete-utopia/utopia/blob/b54495cd1acce119f0274bb4dba7df482b000bd3/utopia-api/src/property-controls/property-controls.ts#L67 )

I think it's less confusing if it's explicit, but I am curious what you and others think